pkgs/development/compilers/mono/default.nix

```diff
@@ -0,0 +1,120 @@
+{ stdenv, lib, fetchurl, fetchpatch, callPackage, cmake, bison, pkgconfig, perl, python2
+, gettext, ncurses, zlib, cacert
+, Foundation, libobjc
+, withX11 ? true, libgdiplus, libX11, libXinerama, cairo, glib, gtk2, gdk_pixbuf, librsvg
+, withCUPS ? true, cups
+, withALSA ? true, alsaLib
+, withLLVM ? true
+}:
```

Comment:
The three options `withCUPS`, `withALSA` and `withLLVM` should default to `false` on Darwin (i.e., using `stdenv.isLinux` or `!stdenv.isDarwin`).